### PR TITLE
fix: Reflect the api for bans

### PR DESCRIPTION
**Pull request by Lulalaby**

### CODE REVIEW

**Comment by Lulalaby** [comment]
Our code/docs are misleading.
See https://discord.com/developers/docs/resources/guild#get-guild-bans-query-string-params

File: discord/guild.py [hunk 1]
```diff
@@ -2156,13 +2156,15 @@ async def fetch_channel(self, channel_id: int, /) -> GuildChannel | Thread:
...
-        before: SnowflakeTime | None = None,
-        after: SnowflakeTime | None = None,
+        before: Snowflake | None = None,
+        after: Snowflake | None = None,
     ) -> BanIterator:
         """|coro|
 
         Retrieves an :class:`.AsyncIterator` that enables receiving the guild's bans. In order to use this, you must
         have the :attr:`~Permissions.ban_members` permission.
+        Provide a user id to before and after for pagination.
```
Comment (Dorukyum) [inline discord/guild.py:2166]:
```suggestion
        Provide a user ID to before and after for pagination.
```

File: discord/guild.py [hunk 2]
```diff
@@ -2174,14 +2176,10 @@ def bans(
...
-        before: Optional[Union[:class:`.abc.Snowflake`, :class:`datetime.datetime`]]
-            Retrieve bans before this date or object.
-            If a datetime is provided, it is recommended to use a UTC aware datetime.
-            If the datetime is naive, it is assumed to be local time.
-        after: Optional[Union[:class:`.abc.Snowflake`, :class:`datetime.datetime`]]
-            Retrieve bans after this date or object.
-            If a datetime is provided, it is recommended to use a UTC aware datetime.
-            If the datetime is naive, it is assumed to be local time.
+        before: Optional[:class:`.abc.Snowflake`]
+            Retrieve bans before given user id.
```
Comment (Dorukyum) [inline discord/guild.py:2180]:
```suggestion
            Retrieve bans before given user ID.
```

File: discord/guild.py [hunk 3]
```diff
@@ -2174,14 +2176,10 @@ def bans(
...
+        after: Optional[:class:`.abc.Snowflake`]
+            Retrieve bans after given user id.
```
Comment (Dorukyum) [inline discord/guild.py:2182]:
```suggestion
            Retrieve bans after given user ID.
```

File: discord/guild.py [hunk 4]
```diff
@@ -2156,13 +2156,15 @@ async def fetch_channel(self, channel_id: int, /) -> GuildChannel | Thread:
...
-        after: SnowflakeTime | None = None,
+        before: Snowflake | None = None,
+        after: Snowflake | None = None,
     ) -> BanIterator:
         """|coro|
 
         Retrieves an :class:`.AsyncIterator` that enables receiving the guild's bans. In order to use this, you must
         have the :attr:`~Permissions.ban_members` permission.
+        Provide a user id to before and after for pagination.
+        Users will always be returned in ascending order by user.id. If both before and after are provided, only before is respected.
```
Comment (Dorukyum) [inline discord/guild.py:2167]:
```suggestion
        Users will always be returned in ascending order sorted by user ID. If both before and after are provided, only before is respected.
```

**Review by Dorukyum — COMMENTED** [review]
A bit of consistency

**Review by Dorukyum — DISMISSED** [review]

**Review by plun1331 — APPROVED** [review]

**Review by Dorukyum — APPROVED** [review]

**Pull request merged**